WooCommerce/WooCommerceTests/ViewRelated/Search/Order/OrderSearchUICommandTests.swift

```diff
@@ -0,0 +1,171 @@
+        // Given
+        let command = OrderSearchUICommand(siteID: siteID, onSelectSearchResult: { _, _ in }, storageManager: storageManager)
+
+        // When
+        let starterViewController = command.createStarterViewController()
+
+        // Then
+        XCTAssertNil(starterViewController, "Expected createStarterViewController to return nil")
+    }
+
+    func testCreateResultsController() {
+        // Given
+        let command = OrderSearchUICommand(siteID: siteID, onSelectSearchResult: { _, _ in }, storageManager: storageManager)
+
+        // When
+        let resultsController = command.createResultsController()
+
+        // Then
+        XCTAssertNotNil(resultsController, "Expected createResultsController to return a ResultsController instance")
+        XCTAssertEqual(resultsController.predicate?.predicateFormat, "siteID == \(siteID)", "Predicate format did not match")
+        XCTAssertEqual(resultsController.sortDescriptors?.first?.key, "dateCreated", "First sort descriptor key did not match")
+        XCTAssertEqual(resultsController.sortDescriptors?.first?.ascending, false, "First sort descriptor ascending did not match")
+    }
+
+    func testCreateCellViewModel() {
+        // Given
+        let mockOrder = MockOrders().makeOrder(status: .onHold, items: [], shippingLines: [], refunds: [], fees: [], taxes: [], customFields: [], giftCards: [])
+
+        let command = OrderSearchUICommand(siteID: siteID, onSelectSearchResult: { _, _ in }, storageManager: storageManager)
+
+        // Insert mock order statuses
+        insertOrderStatuses()
+
+        // When
+        let cellViewModel = command.createCellViewModel(model: mockOrder)
+
+        // Then
+        XCTAssertNotNil(cellViewModel, "Expected createCellViewModel to return an OrderListCellViewModel instance")
+        XCTAssertEqual(cellViewModel.statusString, "on-hold", "Expected order status name to match")
+        XCTAssertEqual(cellViewModel.status, .onHold, "Expected createCellViewModel to return on hold status")
+    }
+
+    func testSanitizeKeyword() {
+        // Given
+        let command = OrderSearchUICommand(siteID: siteID, onSelectSearchResult: { _, _ in }, storageManager: storageManager)
+
+        // When
+        let sanitizedKeywordWithHash = command.sanitizeKeyword("#123")
+        let sanitizedKeywordWithoutHash = command.sanitizeKeyword("123")
+
+        // Then
+        XCTAssertEqual(sanitizedKeywordWithHash, "123", "Expected sanitizeKeyword to remove the leading '#'")
+        XCTAssertEqual(sanitizedKeywordWithoutHash, "123", "Expected sanitizeKeyword to return the keyword unchanged if there's no leading '#'")
+    }
+
+    func testSynchronizeModelsTracksAnalytics() throws {
+        // Given
+        let stores = MockStoresManager(sessionManager: .testingInstance)
+        let command = OrderSearchUICommand(siteID: siteID, onSelectSearchResult: { _, _ in }, storageManager: storageManager, analytics: analytics, stores: stores)
```

Comment:
If you choose to, having this all in properties/setUp makes it a bit easier to read the tests.

Reply:
Prefer to leave it as it is since it improves clarity about the dependencies needed for this functionality. Unlike the other tests this is the only one that requires mocked analytics.